flink-runtime/src/main/java/org/apache/flink/runtime/state/UserFacingListState.java

```diff
@@ -48,6 +49,7 @@
 
 	@Override
 	public void add(T value) throws Exception {
+		Preconditions.checkNotNull(value, "You cannot add null to a ListState.");
```

Comment:
You should also change the `addAll()`  to check for `nulls` internally, as done in the `HeapListState`.

Reply:
Or not change this one, and defer null detection completely to the wrapping state. WDYT?

Reply:
Yes, that would be cleaner.